src/libpgmoneta/walfile/wal_reader.c

```diff
@@ -186,6 +195,18 @@ pgmoneta_wal_parse_wal_file(char* path, struct server* server_info, struct walfi
    uint32_t next_record = ftell(file);
    int page_number = 0;
    wal_file->long_phd = long_header;
+   assert(magic_value_to_postgres_version(long_header->std.xlp_magic) != -1);
+
+   if (server == -1)
```

Comment:
Do we need this ?

Getting the magic number from the file should be enough...

Reply:
I need it because I want to add server info to the shared memory, it's also used outside of the scope of this function, and I need to know whether to write it in the index 0 or another index if it's already defined.

Reply:
That is ok - lets assume that the WAL file "fits"

Reply:
> That is ok - lets assume that the WAL file "fits"

Hmm sorry not sure if I quite understood what you mean.
In general, I need the sever info outside of the scope of this function and I think it's easier to free it when with the shared memory at the end.

Reply:
I mean that the WAL file belongs to the server in question such that the OIDs, relfillnodes, ... match up

Reply:
ok so anything needed to do here?

Reply:
we can always circle back to it